platform-hub-api/app/controllers/kubernetes/clusters_controller.rb

```diff
@@ -1,52 +1,69 @@
 class Kubernetes::ClustersController < ApiJsonController
 
-  before_action :load_kubernetes_clusters_hash_record
+  before_action :find_cluster, only: [ :show, :update ]
+
+  skip_authorization_check only: [ :index, :show ]
+  load_and_authorize_resource class: KubernetesCluster, :except => [:index, :show]
```

Comment:
We still want to authorise the `index` and `show` actions? Also, don't need to use `load_and_authorize_resource` here (unless that's the only one that supports a custom `class`?)

Could change these 4 lines to something like:

```ruby 
before_action :find_cluster, only: [ :show, :update ]

authorize_resource class: KubernetesCluster
```

(assuming that last line actually works).

Reply:
We should probably keep is similar to projects, so index and show don't have to be protected as they never expose sensitive data. Will check if the above works tho.

Reply:
Ah that's a good point – we will probably need these open for the projects self-serve work anyway.